CHAOS-6083 Linux API block fault (#7300)
harness/developer-hub: docs/chaos-engineering/chaos-faults/linux/linux-api-block.md

---
id: linux-api-block
title: Linux API block
---

import Ossupport from './shared/note-supported-os.md'
import FaultPermissions from './shared/fault-permissions.md'

Linux API block injects API block fault into a Linux machine for a specific duration through path filtering. This results in the API not being able to send responses for the requests it receives.

![Linux API block](./static/images/linux-api-block.png)

## Use cases

> - Validates how well your system can handle disruptions in API services for a specific pod.

Reviewer: ```suggestion
- Validates how well your system can handle API service disruptions for a Linux server.
```

Reviewer: Check other places too for pods

      <td> For more information, go to <a href="#advanced-filters">destination hosts</a>.</td>
  </tr>
  <tr>
      <td> statusCode </td>
      <td> Status code received when the API is blocked. </td>
      <td> When the API is blocked, it throws an error along with the status code. Default: 404. </td>
  </tr>
</table>

### Target service port

Port of the target service. Tune it by using the `targetServicePort` input variable.

The following YAML snippet illustrates the use of this input variable:

[embedmd]: # "./static/manifests/linux-api-block/target-service-port.yaml yaml"

```yaml
## provide the port of the target service
apiVersion: litmuschaos.io/v1alpha1
kind: ChaosEngine
metadata:
  name: engine-nginx
spec:
  engineState: "active"
  annotationCheck: "false"
  appinfo:
    appns: "default"
    applabel: "app=nginx"
    appkind: "deployment"
  chaosServiceAccount: litmus-admin
  experiments:
    - name: linux-api-block
      spec:
        components:
          env:
            # provide the port of the target service
            - name: targetServicePort

>               value: "80"

Reviewer: The data types might be different, please check the reference manifests.
```suggestion
              value: 80
```

Reviewer: Check other places as well

            - name: pathFilter
              value: '/status'
            - name: statusCode

>               value: "404"

Reviewer: ```suggestion
              value: 404
```

          env:
            # provide destination ports
            - name: destinationPorts
              value: '80,443'
            # provide the api path filter
            - name: pathFilter
              value: '/status'
            # provide the port of the target service
            - name: targetServicePort
              value: "80"
```

### HTTPS

Enable the HTTPS support for both incoming and outgoing traffic by setting the `httpsEnabled` field to `true`. Its usage varies depending on whether it is applied to `ingress` or `egress` scenario.

#### Ingress

Set this parameter if the HTTPS URL of the target application includes a port whose format is `https://<hostname>:port`. However, if the HTTPS URL is in the format `https://<hostname>` without a port, this setting is not required.

#### Egress

For outbound traffic, set `httpsEnabled` to `true` to enable HTTPS support for external services. This enables using TLS certificates for the proxy within the target application.

- If the HTTP client in the target application is configured to reload certificates with each API call, set `httpsEnabled` to `true`. You won't need to provide `customCertificates` input variable.

> - However, if the root certificate directory and file name differ from `/etc/ssl/certs` and `ca-certificates.crt` respectively, provide the root CA certificate file name using `httpsRootCertFile` input variable.

Reviewer: ```suggestion
- However, if the root CA certificate file path is not `/etc/ssl/certs/ca-certificates.crt`, provide it using `httpsRootCertFile` input variable.
```
